app/code/core/Mage/Adminhtml/controllers/Promo/CatalogController.php

```diff
@@ -308,6 +308,38 @@ public function applyRulesAction()
         $this->_redirect('*/*');
     }
 
+     /**
+     * Duplicate catalog rule and edit
+     */
+    public function duplicateAction()
+    {
+        $errorMessage = Mage::helper('catalogrule')->__('Cannot copy rule ');
+        $params = $this->getRequest()->getParams();
+        try {
+            if($model = Mage::getModel('catalogrule/rule')->load($params['id'])) {
+                $_currentrule = $model->getData();
+                $_newrule = Mage::getModel('catalogrule/rule');
+                $_newrule->setData($_currentrule);
+                unset($_newrule['rule_id']);
+                $_newrule['is_active'] = 0;
+                $_newrule['name'] = $_currentrule['name'] . " (Copy)";
+                $_newrule['identifier'] = $_currentrule['identifier'] . "-copy";
+                unset($_currentrule);
+                $_newrule->save();
+                $this->_getSession()->addSuccess(Mage::helper('catalogrule')->__('The rule has been duplicated. Create a new Rule Name and Save to apply changes.'));
+                $this->_redirect('*/*/edit', array('id' => $_newrule->getId(), '_current' => true));
```

Comment:
I think you're going about this the wrong way.. Rather than duplicate and edit, I think it is best to just preload the form data from an existing rule so that the rule is not created until it is actually saved. In the case of rules especially they can cause quite a bit of work by indexers and such when they are saved, cache invalidations, etc. To do this you would probably copy or modify the `editAction`.